src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -607,12 +613,147 @@ CHIP_ERROR DefaultDACVerifier::VerifyNodeOperationalCSRInformation(const ByteSpa
     return CHIP_NO_ERROR;
 }
 
+// This method parses the below JSON Scheme
+// [
+//  {
+//    "type": "revocation_set",
+//    "issuer_subject_key_id": "63540E47F64B1C38D13884A462D16C195D8FFB3C",
+//    "issuer_name": "MD0xJTAjBgNVBAMMHE1hdHRlciBEZXYgUEFJIDB4RkZGMSBubyBQSUQxFDASBgorBgEEAYKifAIBDARGRkYx",
+//    "revoked_serial_numbers": [
+//      "69CDF10DE9E54ED1"
+//    ]
+//  }
+// ]
+//
+bool DefaultDACVerifier::IsEntryExistsInRevocationSet(const CharSpan & akidHexStr, const CharSpan & issuerNameBase64Str,
+                                                      const CharSpan & serialNumberHexStr)
+{
+    std::ifstream file(mDeviceAttestationRevocationSetPath);
+    if (!file.is_open())
+    {
+        return false;
+    }
+
+    // Parse the JSON data incrementally
+    Json::CharReaderBuilder readerBuilder;
+    Json::Value jsonData;
+    std::string errs;
+
+    bool parsingSuccessful = Json::parseFromStream(readerBuilder, file, &jsonData, &errs);
+
+    // Close the file as it's no longer needed
+    file.close();
+
+    if (!parsingSuccessful)
+    {
+        return false;
+    }
+
+    for (const auto & revokedSet : jsonData)
+    {
+        if (strncmp(revokedSet["issuer_name"].asCString(), issuerNameBase64Str.data(), issuerNameBase64Str.size()) == 0 &&
+            strncmp(revokedSet["issuer_subject_key_id"].asCString(), akidHexStr.data(), akidHexStr.size()) == 0)
+        {
+            for (const auto & revokedSerialNumber : revokedSet["revoked_serial_numbers"])
+            {
+                if (strncmp(revokedSerialNumber.asCString(), serialNumberHexStr.data(), serialNumberHexStr.size()) == 0)
+                {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+CHIP_ERROR DefaultDACVerifier::GetAKIDHexStr(const ByteSpan & certDer, MutableCharSpan & outAKIDHexStr)
+{
+    uint8_t akidBuf[kAuthorityKeyIdentifierLength];
+    MutableByteSpan akid(akidBuf);
+
+    CHIP_ERROR err = ExtractAKIDFromX509Cert(certDer, akid);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+    VerifyOrReturnError(outAKIDHexStr.size() > akid.size() * 2, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    Encoding::HexFlags flags = Encoding::HexFlags::kUppercaseAndNullTerminate;
+    err                      = BytesToHex(akid.data(), akid.size(), outAKIDHexStr.data(), outAKIDHexStr.size(), flags);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+    outAKIDHexStr.reduce_size(strlen(outAKIDHexStr.data()));
```

Comment:
This makes no sense, but does explain why we were null-terminating...

The size must be `akid.size() * 2`, no?